### ISSUE TITLE
Can't run on other thread

### ISSUE THREAD

**Issue by dominic7125** [SciSharp/Keras.NET#77]
Does it only can run on main thread?
If so, it may stuck GUI while fitting.
  
`Console.WriteLine("start");

            //Load train data
            NDarray x = np.array(new float[,] { { 0, 0 }, { 0, 1 }, { 1, 0 }, { 1, 1 } });
            NDarray y = np.array(new float[] { 0, 1, 1, 0 });

            Task.Run(() =>
            {
                //Build sequential model
                model = new Sequential();
                model.Add(new Dense(32, activation: "relu", input_shape: new Shape(2)));
                model.Add(new Dense(64, activation: "relu"));
                model.Add(new Dense(1, activation: "sigmoid"));

                var earlystop = new EarlyStopping(monitor: "val_loss", min_delta: 0.1f, patience: 500, verbose: 1, mode: "auto", baseline: 0, restore_best_weights: false);
                
                //Compile and train
                model.Compile(optimizer: "sgd", loss: "binary_crossentropy", metrics: new string[] { "accuracy" });
                var history = model.Fit(x, y, batch_size: 2, epochs: 1000, verbose: 1, callbacks: new Callback[] { earlystop });
            }).Wait();

            Console.WriteLine("end");`

**Comment by newrohlzy** [comment]
hi,I encountered a problem when I tried to run the Demo in the home page. I wonder if you have run successfully?I will post the error I encountered, if you know how to solve it, please help me, thank you.
The reasons for the error are as follows:
"System.DllNotFoundException:“Unable to load DLL 'python36' or one of its dependencies: 找不到指定的模块。 (Exception from HRESULT: 0x8007007E)”
"

**Comment by henon** [comment]
@dominic7125 read this howto about Numpy.NET and multi-threading: https://github.com/SciSharp/Numpy.NET#multi-threading

**Comment by formatx** [comment]
Hi @henon , I have a similiar question.
Im quite new to whole python\Keras, and trying to figure out if I`m able to load a model and then run several predictions in parallel on it.
So far I managed to avoid exceptions thanks to the info in the link you provided:

using (Py.GIL())
            {
                NDarray npArray = np.array(input);
                NDarray prediction = _model.Predict(npArray);
                result = (double)prediction[0, 0];
            }
            return result;

However it runs even slower than what I get when running it one by one.
I found that theres something like 
`model._make_predict_function()`
[https://github.com/keras-team/keras/issues/6124](url)
in Keras, but I didn`t find it in Keras.Net, and still not sure that compiling predict function would help me with parallelism.

So my question is: Is it really possible to run predictions in parallel using Keras.Net otherwise than starting two separate processes?
